src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpDynamicInterfaceCastableImplementation.Fixer.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.NetCore.Analyzers.InteropServices;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.InteropServices
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public sealed class CSharpDynamicInterfaceCastableImplementationFixer : DynamicInterfaceCastableImplementationFixer
+    {
+        protected override async Task<Document> ImplementInterfacesOnDynamicCastableImplementation(
+            SyntaxNode declaration,
+            Document document,
+            CancellationToken ct)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, ct).ConfigureAwait(false);
+
+            INamedTypeSymbol type = (INamedTypeSymbol)editor.SemanticModel.GetDeclaredSymbol(declaration, ct);
+            var generator = editor.Generator;
+
+            var defaultMethodBodyStatements = generator.DefaultMethodBody(editor.SemanticModel.Compilation).ToArray();
+            List<SyntaxNode> generatedMembers = new List<SyntaxNode>();
+            foreach (var iface in type.AllInterfaces)
```

Comment:
Are there any tests that will fail if `AllInterfaces` is replaced with `Interfaces`? If not, considering adding one.

Reply:
Yes, the `DynamicInterfaceCastableImplementation_GrandparentInterfaceMethodsNotImplementated_CS_Diagnostic` test would fail if we used `Interfaces` instead of `AllInterfaces`.